source/extensions/filters/common/ext_authz/check_request_utils.cc

```diff
@@ -145,15 +146,18 @@ void CheckRequestUtils::setHttpRequest(
   // Calling mutable_header_map() creates the field in the request; only do so when necessary.
   auto* mutable_header_map = encode_raw_headers ? httpreq.mutable_header_map() : nullptr;
 
-  headers.iterate([encode_raw_headers, request_header_matchers, mutable_headers,
-                   mutable_header_map](const Envoy::Http::HeaderEntry& e) {
+  headers.iterate([encode_raw_headers, allowed_headers_matcher, disallowed_headers_matcher,
+                   mutable_headers, mutable_header_map](const Envoy::Http::HeaderEntry& e) {
     // Skip any client EnvoyAuthPartialBody header, which could interfere with internal use.
     if (e.key().getStringView() == Headers::get().EnvoyAuthPartialBody.get()) {
       return Envoy::Http::HeaderMap::Iterate::Continue;
     }
 
     const std::string key(e.key().getStringView());
-    if (request_header_matchers != nullptr && !request_header_matchers->matches(key)) {
+    if (allowed_headers_matcher != nullptr && !allowed_headers_matcher->matches(key)) {
+      return Envoy::Http::HeaderMap::Iterate::Continue;
+    }
+    if (disallowed_headers_matcher != nullptr && disallowed_headers_matcher->matches(key)) {
```

Comment:
it seems that we can do following to emphasize that `disallowed_headers` take precedence?

```
if (disallowed_headers_matcher != nullptr && disallowed_headers_matcher->matches(key)) {
  continue;
}

if (allowed_headers_matcher != nullptr && !allowed_headers_matcher->matches(key)) {
      return Continue;
 }
```

Reply:
Agreed, done.